rabbitmq/src/urabbitmq/impl/amqp_connection_handler.cpp

```diff
@@ -45,14 +46,54 @@ engine::io::Socket CreateSocket(clients::dns::Resolver& resolver,
 
 std::unique_ptr<engine::io::RwBase> CreateSocketPtr(
     clients::dns::Resolver& resolver, const AMQP::Address& address,
-    engine::Deadline deadline) {
+    const AuthSettings& auth_settings, engine::Deadline deadline) {
   auto socket = CreateSocket(resolver, address, deadline);
 
   const bool secure = address.secure();
   if (secure) {
+    if (!auth_settings.client_cert_path.empty() ||
+        !auth_settings.client_private_key_path.empty() ||
+        !auth_settings.ca_cert_paths.empty()) {
+      if (auth_settings.client_cert_path.empty() !=
+          auth_settings.client_private_key_path.empty()) {
+        throw std::runtime_error(
+            "Either set both tls.client-cert-path and "
+            "tls.client-private-key-path options or none of "
+            "them");
+      }
+
+      crypto::Certificate client_cert;
+      if (!auth_settings.client_cert_path.empty()) {
+        auto contents =
+            fs::blocking::ReadFileContents(auth_settings.client_cert_path);
```

Comment:
I think  `fs::` operations should be performed once, at the component initialization, for two reasons:
1. They are partly a validation
2. They are potentially costly with a high socket reopen rate (shouldn't be a common case. but still)

Reply:
1. I thought it will be a good idea to (re)load them in runtime in case certs will change on disk (as a part of regular certs rotation for example) without service restart.
2. Even if there are a high socket reopen rate, read should be relatively fast once files will be in filesystem cache.

Reply:
Ok, fixed